code/modules/mob_spawn/mob_spawn.dm

```diff
@@ -231,8 +248,38 @@
 	if(!mob_possessor.key) // This is in the scenario that the server is somehow lagging, or someone fucked up their code, and we try to spawn the same person in twice. We'll simply not spawn anything and CRASH(), so that we report what happened.
 		CRASH("Attempted to create an instance of [type] with a mob that had no ckey attached to it, which isn't supported by ghost role spawners!")
 
-	return ..()
+	var/load_prefs = FALSE
+	//if we can load our own appearance and its not restricted, try
+	if(!random_appearance && mob_possessor?.client)
+		//if we have gotten to this point, they have already waived their species pref.-- they were told they need to use the specific species already
+		if((restricted_species && (mob_possessor?.client?.prefs?.read_preference(/datum/preference/choiced/species) in restricted_species)) || !restricted_species)
+			var/appearance_choice = tgui_alert(mob_possessor, "Use currently loaded character preferences?", "Appearance Type", list("Yes", "No"))
+			if(appearance_choice == "Yes")
+				load_prefs = TRUE
+
+	var/mob/living/carbon/human/spawned_human = ..(mob_possessor, newname, load_prefs)
+
+	if(!load_prefs)
+		var/datum/language_holder/holder = spawned_human.get_language_holder()
+		holder.get_selected_language() //we need this here so a language starts off selected
```

Comment:
```suggestion
		spawned_human.get_language_holder().get_selected_language() //we need this here so a language starts off selected
```

Reply:
may need to pull the latest master if this results in the compiler or linters complaining